overlord/snapstate/snapstate.go

```diff
@@ -626,7 +668,10 @@ func doInstall(st *state.State, snapst *SnapState, snapsup *SnapSetup, flags int
 	// additional tasks.
 	linkSnap := st.NewTask("link-snap", fmt.Sprintf(i18n.G("Make snap %q%s available to the system"), snapsup.InstanceName(), revisionStr))
 	addTask(linkSnap)
-	prev = linkSnap
+
+	for _, t := range tasksAfterLinkSnap {
+		addTask(t)
```

Comment:
does this match where we specced after link hooks for components should run vs snap own hooks?

Reply:
The component hook tasks aren't created yet. Should I add a TODO here for once https://github.com/snapcore/snapd/pull/13775 gets merged?

Reply:
As discussed, we are not adding hook tasks yet in `doInstallComponent()`, but that will eventually happen. In any case, if `componentTasksForInstallWithSnap()` keeps the tasks order things should be fine, maybe for the moment we could just add a comment there commenting that order needs to be kept.